webapp/src/types/store.ts

```diff
@@ -0,0 +1,26 @@
+import {GlobalState as ReduxGlobalState} from 'mattermost-redux/types/store';
+
+import {Item} from './gitlab_items';
+
+import {GitlabUsersData, LHSData, ShowRhsPluginActionData, UserSettingsData} from '.';
+
+export type GlobalState = ReduxGlobalState & {
+    'plugins-com.github.manland.mattermost-plugin-gitlab': PluginState
+};
+
+export type PluginState = {
+    connected: boolean,
+    gitlabURL: string,
+    organization: string,
+    username: string,
+    reviewDetails: Item[],
+    yourPrDetails: Item[],
+    lhsData: LHSData,
+    rhsState: string,
+    settings: UserSettingsData,
+    gitlab_client_id: string;
+    gitlabUsers: Record<string, GitlabUsersData>,
+    rhsPluginAction: ShowRhsPluginActionData,
+};
```

Comment:
Is there a way we can have this generated from the exported `combineReducers` call in the reducers file? Then we know the store is based off of the reducers, and that the reducers are correctly typed.